gtdynamics/universal_robot/Robot.cpp

```diff
@@ -82,8 +82,10 @@ Robot Robot::fixLink(const std::string &name) {
   if (name_to_link_.find(name) == name_to_link_.end()) {
     throw std::runtime_error("no link named " + name);
   }
-  name_to_link_.at(name)->fix();
-  return *this;
+
+  Robot fixed_robot = Robot(*this);
+  fixed_robot.name_to_link_.at(name)->fix();
```

Comment:
so, is fix now private? Otherwise this will become an issue again :-)

Reply:
Ah good catch.

Reply:
Okay so `sdf.cpp` needs the `fix` method, so I'm making both `Link::fix` and `Link::unfix` functional as well.

Reply:
I figured out a cleaner way.